### DISCUSSION TITLE
Is there a equivalent keyword for `diff_files` before version 0.63.0

### DISCUSSION THREAD

**Discussion by MichaelJeiShen** [mesonbuild/meson#10660] [Q&A] [answered]
Hi all,

I am trying to apply a diff file on my `subprojects`.
I know that on 0.63.0 there is a new feature [diff_files](https://mesonbuild.com/Wrap-dependency-system-manual.html#diff-files) can achive what I said.

However I need to stick with **0.61.3** for some reasons.
Is there a way to fulfill my needs in **0.61.3**?

Thanks,
Michael

**Comment by eli-schwartz** [comment, marked as answer]
You can directly fork the subproject, and use the fork instead.

**Reply by MichaelJeiShen** [reply]
You mean manually fork it to `subprojects` then apply the diff files?
```
git clone <url> subprojects/projA
cd subprojects/projA
git am <diff file>
```

I can't do this becuase our convention is to rely on `.wrap` file to clone and build subproject.
Adding entire repo to `subprojects` is not allowed.

**Reply by eli-schwartz** [reply]
That's why diff_files was added, because it offers functionality not otherwise available. :)

You could fork https://example.org/~author/projA.git to https://example.org/~MichaelJeiShen/projA-patched.git, and use the latter in your .wrap file; this is what people did before Meson 0.63.0

**Reply by MichaelJeiShen** [reply]
> You could fork https://example.org/~author/projA.git to https://example.org/~MichaelJeiShen/projA-patched.git, and use the latter in your .wrap file; this is what people did before Meson 0.63.0

I afraid I can't do that neither..
We only allow to use trusted git repo.
I am sure my personal github is not qualified.

> That's why diff_files was added, because it offers functionality not otherwise available. :)

I see.. Thanks for explaining :)

Mark as closed.

**Reply by eli-schwartz** [reply]
> We only allow to use trusted git repo.

In general I think this is a good idealistic policy too. :) I'm sorry that there's no awesome solution for you, but hopefully you can find a way to upgrade to 0.63.0 and use the cool new feature.

**Reply by simgt** [reply]
Thanks for your post. I struggled for a while because meson wasn't applying the patch, without complaining about not knowing what to do with `diff_files`. I clearly didn't pay attention to the mention of `Since: 0.63.0` in the manual...

> I afraid I can't do that neither.. We only allow to use trusted git repo. I am sure my personal github is not qualified.

I'm a bit late to the game but it may helps others: the solution in your case is likely to just squash the repo in a subproject with git subtree. It would also allow you to push back changes upstream without forking.
